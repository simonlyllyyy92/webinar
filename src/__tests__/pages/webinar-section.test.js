import React from "react"

import { shallow } from "enzyme"

import WebinarSection from "../../components/webinar-section/webinar-section"
import toJson from "enzyme-to-json"

test("should render webinar section with cards", () => {
  const demoData = {
    loadingStatus: false,
    favorited: true,
    handleScroll: jest.fn(),
    webinarList: [
      {
        id: "123",
        favourited: true,
        created_at: "2020",
        title: "this is the title",
        content: "this is the content",
      },
      {
        id: "125",
        favourited: true,
        created_at: "2020",
        title: "this is the title",
        content: "this is the content",
      },
      {
        id: "124",
        favourited: false,
        created_at: "2020",
        title: "this is the title",
        content: "this is the content",
      },
    ],
  }
  const wrapper = shallow(<WebinarSection {...demoData} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
