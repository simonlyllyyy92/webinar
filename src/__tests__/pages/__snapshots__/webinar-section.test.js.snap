// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render webinar section with cards 1`] = `
<div
  className="webinar-section"
>
  <div
    className="webinar-card-container"
    onScroll={[MockFunction]}
  >
    <withRouter(Card)
      content="this is the content"
      created_at="2020"
      id="123"
      key="123"
      title="this is the title"
      unregister={true}
    />
    <withRouter(Card)
      content="this is the content"
      created_at="2020"
      id="125"
      key="125"
      title="this is the title"
      unregister={true}
    />
  </div>
</div>
`;
